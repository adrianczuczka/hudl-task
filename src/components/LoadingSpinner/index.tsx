import React from 'react';
import Spinner from 'react-bootstrap/cjs/Spinner';

import styles from './styles.module.scss';

type Props = {
  centered?: boolean;
};

const LoadingSpinner = ({ centered = false }: Props) => {
  if (centered) {
    return (
      <Spinner animation="border" role="status" className={styles.centered}>
        <span className="sr-only">Loading...</span>
      </Spinner>
    );
  }
  return (
    <Spinner animation="border" role="status">
      <span className="sr-only">Loading...</span>
    </Spinner>
  );
};

export default LoadingSpinner;
