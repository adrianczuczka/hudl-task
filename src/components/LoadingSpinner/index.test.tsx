import React from 'react';
import renderer from 'react-test-renderer';
import LoadingSpinner from './index';

it('renders correctly', () => {
  const tree = renderer.create(<LoadingSpinner />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when centered', () => {
  const tree = renderer.create(<LoadingSpinner centered />).toJSON();
  expect(tree).toMatchSnapshot();
});
