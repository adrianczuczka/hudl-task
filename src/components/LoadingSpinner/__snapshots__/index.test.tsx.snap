// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="spinner-border"
  role="status"
>
  <span
    className="sr-only"
  >
    Loading...
  </span>
</div>
`;

exports[`renders correctly when centered 1`] = `
<div
  className="centered spinner-border"
  role="status"
>
  <span
    className="sr-only"
  >
    Loading...
  </span>
</div>
`;
